pareto/utilities/process_data.py

```diff
@@ -757,59 +762,82 @@ def total_pw_capacity_rule(model, t):
                 )
                 for r in model.s_R
             )
+            * model.model_units["volume"]
+            + sum(
+                model.p_sigma_Storage[s]
+                + _get_max_value_for_parameter(model.p_delta_Storage)
+                for s in model.s_S  # Storage
+            )
         )  # Treatment: for each treatment site, select treatment technology that yields the
         # maximum value for (initial treatment + max treatment expansion)
 
-    strategic_model.e_sigma_MaxPWCapacity = Expression(
+    strategic_model.e_MaxPWCapacity = Expression(
         strategic_model.s_T,
         rule=total_pw_capacity_rule,
         doc="Combined produced water capacity in system [volume]",
     )
 
     def total_water_demand_rule(model, t):
         # Note: if completions pad is outside system, demand is not required to be met
-        return sum(
-            model.p_gamma_Completions[cp, t]
-            if model.p_chi_OutsideCompletionsPad[cp] == 0
-            else 0
-            for cp in model.s_CP
+        return (
+            sum(
+                model.p_gamma_Completions[cp, t]
+                if model.p_chi_OutsideCompletionsPad[cp] == 0
+                else 0
+                for cp in model.s_CP
+            )
+            * model.model_units["time"]
         )
 
-    strategic_model.e_gamma_TimePeriodDemand = Expression(
+    strategic_model.e_TimePeriodDemand = Expression(
         strategic_model.s_T,
         rule=total_water_demand_rule,
         doc="Total water demand at each time period [volume]",
     )
 
     def total_water_available_rule(model, t):
-        return model.e_beta_TotalPW[t] + sum(
-            model.p_sigma_ExternalWater[f, t] for f in model.s_F
+        return (
+            model.e_TotalPW[t]
+            + sum(model.p_sigma_ExternalWater[f, t] for f in model.s_F)
```

Comment:
Should we not add the stored water for the water available rule?

Reply:
@JavalVyas2000 the issue is that this check runs before solving the model, and it's not known a priori how much water will be in storage because that is a decision variable for the model. You are right that it could be possible to use stored water to meet demand at a time where external + produced + flowback water volumes are not sufficient, which this check could potentially miss. I will think about how to change this check.

Reply:
Oh ok got it. We may want to have this as an issue for further enhancement?

Reply:
Yeah, you are right that it's probably best to open a new issue for this rather than addressing it in this PR. I'll open an issue later and link to it here.

Reply:
I've opened issue #329 to address this in the future.